YesGraph/yesgraph-superwidget
# Automatic versioning

#### What’s this PR do?
- Adds a `gulp version` task, which updates version numbers on each file (sdk, superwidget, & css) based on the update type (`patch`, `minor`, or `major`).
- Splits the `gulp` tasks into separate files.

#### Any background context you want to provide?
Manual updates to the version numbers was time consuming and error-prone. These changes get us closer to a fully automated deploy process. Also, we hadn't specified how our versioning system should work until now. We will use [Semantic Versioning](http://semver.org/#summary), which works like this:

Given a version number `MAJOR.MINOR.PATCH`, increment the:
- `MAJOR` version when you make incompatible API changes,
- `MINOR` version when you add functionality in a backwards-compatible manner, and
- `PATCH` version when you make backwards-compatible bug fixes.

#### Where should the reviewer start?
gulptools/version.js

#### Does this require changes on the API side?
Nope

#### How should this be manually tested?
- Look at gulptools/config.js. You should see the version for each file:
 <img width="366" alt="screen shot 2016-06-13 at 11 28 21 am" src="https://cloud.githubusercontent.com/assets/10701968/16018699/71d202be-315a-11e6-8b91-7b7d37b5953a.png">
- Then run `gulp`. When prompted, choose an update type for the sdk, the superwidget, and the css.
- Now check that the version numbers in the config file (gulptools/config.js) have changed.
- Lastly, run `git checkout HEAD -- gulptools/config.js` to reset the version numbers.

#### What are the relevant tickets?
https://app.asana.com/0/77757700101210/142347704015138
